ddtrace/tracer/sampler.go

```diff
@@ -185,42 +186,51 @@ func newRulesSampler(rules []SamplingRule) *rulesSampler {
 	}
 }
 
-// appliedSamplingRules validates the user-provided rules and returns an internal representation.
-// If the DD_TRACE_SAMPLING_RULES environment variable is set, it will replace the given rules.
-func appliedSamplingRules(rules []SamplingRule) []SamplingRule {
+// samplingRulesFromEnv parses sampling rules from the DD_TRACE_SAMPLING_RULES
+// environment variable.
+func samplingRulesFromEnv() ([]SamplingRule, error) {
 	rulesFromEnv := os.Getenv("DD_TRACE_SAMPLING_RULES")
-	if rulesFromEnv != "" {
-		rules = rules[:0]
-		jsonRules := []struct {
-			Service string      `json:"service"`
-			Name    string      `json:"name"`
-			Rate    json.Number `json:"sample_rate"`
-		}{}
-		err := json.Unmarshal([]byte(rulesFromEnv), &jsonRules)
+	if rulesFromEnv == "" {
+		return nil, nil
+	}
+	jsonRules := []struct {
+		Service string      `json:"service"`
+		Name    string      `json:"name"`
+		Rate    json.Number `json:"sample_rate"`
+	}{}
+	err := json.Unmarshal([]byte(rulesFromEnv), &jsonRules)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing DD_TRACE_SAMPLING_RULES: %v", err)
+	}
+	rules := make([]SamplingRule, 0, len(jsonRules))
+	var errStr string
+	for _, v := range jsonRules {
+		if v.Rate == "" {
+			errStr += "rate not provided "
+			continue
```

Comment:
As soon as something goes wrong, let's return with an error
```suggestion
			return nil, fmt.Errorf("rule at index %d has no rate", i)
```

Reply:
This would not be backwards-compatible, as some users may be running with some invalid rules.
Currently, the valid rules would still take effect. If we change this to return an error, the first invalid rule would cause none of the others to take effect.